pyccel/ast/operators.py

```diff
@@ -384,6 +384,7 @@ def _calculate_dtype(cls, *args):
         e.g.
             1 + 2j -> PyccelAdd(LiteralInteger, LiteralComplex) -> complex
         """
+
         integers  = [a for a in args if a.dtype in (NativeInteger(),NativeBool())]
         floats    = [a for a in args if a.dtype is NativeFloat()]
```

Comment:
Are the list comprehension necessary, it doesn't look like they'll be used later, wouldn't it be better to just use `args[0].dtype`?

Reply:
For example `3 + 4.0` will lead to a float with the current implementation, but would give an integer with your suggestion